File: app/src/main/java/com/veganbeauty/app/features/home/BottomNavHelper.java
package com.veganbeauty.app.features.home;

import android.animation.ArgbEvaluator;
import android.animation.ValueAnimator;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.graphics.drawable.ColorDrawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.DecelerateInterpolator;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;
import androidx.interpolator.view.animation.FastOutSlowInInterpolator;

import com.veganbeauty.app.R;
import com.veganbeauty.app.data.local.ProfileSession;
import com.veganbeauty.app.features.community.beauty_hub.CommunityBeautyHubFragment;
import com.veganbeauty.app.features.community.com_feed.ComLoadingFragment;
import com.veganbeauty.app.features.home.welcome.HomeWelcomeActivity;
import com.veganbeauty.app.features.myskin.MySkinFragment;
import com.veganbeauty.app.features.profile.AccountProfileFragment;
import com.veganbeauty.app.features.shop.home.ShopHomeFragment;
import com.veganbeauty.app.ui.widget.NavAiSkinGlowAnimator;

public class BottomNavHelper {

    private static final int ACTIVE_COLOR = Color.parseColor("#3E4D44");
    private static final int INACTIVE_COLOR = Color.parseColor("#8FA882");
    private static final float ACTIVE_SCALE = 1.12f;
    private static final long ANIM_DURATION_MS = 220L;

    private static final int[] TAB_IDS = {
            R.id.nav_home,
            R.id.nav_shop,
            R.id.nav_myskin,
            R.id.nav_community,
            R.id.nav_account
    };

    public interface OnTabSelectedListener {
        void onTabSelected(int tabId);
    }

    public static void setup(Fragment fragment, View root, int activeTabId, OnTabSelectedListener onTabSelected) {
        for (int viewId : TAB_IDS) {
            ViewGroup tab = root.findViewById(viewId);
            if (tab == null) {
                continue;
            }
            tab.setOnClickListener(v -> {
                if (viewId == activeTabId) {
                    return;
                }
                if (!ProfileSession.isLoggedIn(root.getContext()) && viewId == R.id.nav_account) {
                    showLoginRequiredDialog(root.getContext());
                    return;
                }
                if (onTabSelected != null) {
                    onTabSelected.onTabSelected(viewId);
                }
            });
        }

        highlightTab(root, activeTabId, false);

        View glowView = root.findViewById(R.id.view_ai_skin_glow);
        View mySkinTab = root.findViewById(R.id.nav_myskin);
        if (glowView != null && mySkinTab != null) {
            NavAiSkinGlowAnimator.attach(glowView, mySkinTab);
        }
    }

    public static void highlightTab(View root, int activeTabId) {
        highlightTab(root, activeTabId, true);
    }

    public static void highlightTab(View root, int activeTabId, boolean animate) {
        for (int tabId : TAB_IDS) {
            ViewGroup tab = root.findViewById(tabId);
            if (tab == null) {
                continue;
            }

            if (tabId == R.id.nav_myskin) {
                ImageView icon = tab.findViewById(R.id.ivMySkin);
                if (icon != null) {
                    icon.clearColorFilter();
                    icon.setImageResource(R.drawable.ic_skin_mainbar_nav);
                }
                continue;
            }

            boolean isActive = tabId == activeTabId;
            ImageView icon = tab.findViewById(getIconId(tabId));
            TextView label = tab.findViewById(getLabelId(tabId));
            applyTabVisualState(icon, label, isActive, animate);
        }
    }

    private static void applyTabVisualState(ImageView icon, TextView label, boolean isActive, boolean animate) {
        int targetColor = isActive ? ACTIVE_COLOR : INACTIVE_COLOR;
        float targetScale = isActive ? ACTIVE_SCALE : 1f;

        if (icon != null) {
            if (animate) {
                animateColor(icon, isActive);
                icon.animate()
                        .scaleX(targetScale)
                        .scaleY(targetScale)
                        .setDuration(ANIM_DURATION_MS)
                        .setInterpolator(new FastOutSlowInInterpolator())
                        .start();
            } else {
                icon.setColorFilter(targetColor);
                icon.setScaleX(targetScale);
                icon.setScaleY(targetScale);
            }
        }

        if (label != null) {
            if (animate) {
                animateTextColor(label, isActive);
                label.animate()
                        .scaleX(isActive ? 1.04f : 1f)
                        .scaleY(isActive ? 1.04f : 1f)
                        .setDuration(ANIM_DURATION_MS)
                        .setInterpolator(new DecelerateInterpolator())
                        .start();
            } else {
                label.setTextColor(targetColor);
                label.setScaleX(isActive ? 1.04f : 1f);
                label.setScaleY(isActive ? 1.04f : 1f);
            }
            label.setTypeface(null, isActive ? Typeface.BOLD : Typeface.NORMAL);
        }
    }

    private static void animateColor(ImageView icon, boolean becomingActive) {
        int from = becomingActive ? INACTIVE_COLOR : ACTIVE_COLOR;
        int to = becomingActive ? ACTIVE_COLOR : INACTIVE_COLOR;
        ValueAnimator animator = ValueAnimator.ofObject(new ArgbEvaluator(), from, to);
        animator.setDuration(ANIM_DURATION_MS);
        animator.setInterpolator(new FastOutSlowInInterpolator());
        animator.addUpdateListener(a -> icon.setColorFilter((int) a.getAnimatedValue()));
        animator.start();
    }

    private static void animateTextColor(TextView label, boolean becomingActive) {
        int from = becomingActive ? INACTIVE_COLOR : ACTIVE_COLOR;
        int to = becomingActive ? ACTIVE_COLOR : INACTIVE_COLOR;
        ValueAnimator animator = ValueAnimator.ofObject(new ArgbEvaluator(), from, to);
        animator.setDuration(ANIM_DURATION_MS);
        animator.setInterpolator(new FastOutSlowInInterpolator());
        animator.addUpdateListener(a -> label.setTextColor((int) a.getAnimatedValue()));
        animator.start();
    }

    private static int getLabelId(int tabId) {
        if (tabId == R.id.nav_home) return R.id.tv_nav_home;
        if (tabId == R.id.nav_shop) return R.id.tv_nav_shop;
        if (tabId == R.id.nav_community) return R.id.tv_nav_community;
        if (tabId == R.id.nav_account) return R.id.tv_nav_account;
        return 0;
    }

    private static int getIconId(int tabId) {
        if (tabId == R.id.nav_home) return R.id.iv_nav_home;
        if (tabId == R.id.nav_shop) return R.id.iv_nav_shop;
        if (tabId == R.id.nav_community) return R.id.iv_nav_community;
        if (tabId == R.id.nav_account) return R.id.iv_nav_account;
        return 0;
    }

    public static void showLoginRequiredDialog(Context context) {
        View dialogView = LayoutInflater.from(context).inflate(R.layout.dialog_login_required, null);
        AlertDialog dialog = new AlertDialog.Builder(context)
                .setView(dialogView)
                .create();

        View btnConfirm = dialogView.findViewById(R.id.btnConfirmLogin);
        View btnCancel = dialogView.findViewById(R.id.btnCancelLogin);

        if (btnConfirm != null) {
            btnConfirm.setOnClickListener(v -> {
                dialog.dismiss();
                Intent intent = new Intent(context, HomeWelcomeActivity.class);
                intent.putExtra("DIRECT_LOGIN", true);
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                context.startActivity(intent);
            });
        }

        if (btnCancel != null) {
            btnCancel.setOnClickListener(v -> dialog.dismiss());
        }

        if (dialog.getWindow() != null) {
            dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        }
        dialog.show();
    }

    public static void navigate(Fragment fragment, int tabId) {
        boolean isLoggedIn = ProfileSession.isLoggedIn(fragment.requireContext());
        Fragment target = null;

        if (tabId == R.id.nav_home) {
            target = new HomeFragment();
        } else if (tabId == R.id.nav_shop) {
            target = new ShopHomeFragment();
        } else if (tabId == R.id.nav_myskin) {
            target = new MySkinFragment();
        } else if (tabId == R.id.nav_community) {
            target = isLoggedIn ? new ComLoadingFragment() : new CommunityBeautyHubFragment();
        } else if (tabId == R.id.nav_account) {
            target = new AccountProfileFragment();
        }

        if (target != null) {
            View root = fragment.getView();
            if (root != null) {
                View navRoot = root.findViewById(R.id.bottom_nav);
                if (navRoot != null) {
                    highlightTab(navRoot, tabId, true);
                }
            }
            fragment.getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, target)
                    .commitAllowingStateLoss();
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/home/BottomNavHelper.kt
package com.veganbeauty.app.features.home

import android.graphics.Color
import android.graphics.Typeface
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.veganbeauty.app.R
import com.veganbeauty.app.features.profile.AccountProfileFragment
import com.veganbeauty.app.features.shop.product.list.ShopListFragment

object BottomNavHelper {

  private const val ACTIVE_COLOR = "#677559"
  private const val INACTIVE_COLOR = "#DDDFC4"

  fun setup(
      fragment: Fragment,
      root: View,
      activeTabId: Int,
      onTabSelected: (Int) -> Unit
  ) {
    val tabs = listOf(
        R.id.nav_home to R.id.nav_home,
        R.id.nav_shop to R.id.nav_shop,
        R.id.nav_myskin to R.id.nav_myskin,
        R.id.nav_community to R.id.nav_community,
        R.id.nav_account to R.id.nav_account
    )

    tabs.forEach { (viewId, _) ->
      root.findViewById<android.view.ViewGroup>(viewId)?.setOnClickListener {
        if (viewId != activeTabId) {
            val isLoggedIn = com.veganbeauty.app.data.local.ProfileSession.isLoggedIn(root.context)
            if (!isLoggedIn && viewId == R.id.nav_account) {
                showLoginRequiredDialog(root.context)
                return@setOnClickListener
            }
            onTabSelected(viewId)
        }
      }
    }

    highlightTab(root, activeTabId)
  }

  fun highlightTab(root: View, activeTabId: Int) {
    val tabIds = listOf(
        R.id.nav_home,
        R.id.nav_shop,
        R.id.nav_myskin,
        R.id.nav_community,
        R.id.nav_account
    )

    tabIds.forEach { tabId ->
      val tab = root.findViewById<android.view.ViewGroup>(tabId) ?: return@forEach
      val icon = tab.getChildAt(0) as? ImageView
      val label = tab.getChildAt(1) as? TextView
      val isActive = tabId == activeTabId
      val color = if (isActive) ACTIVE_COLOR else INACTIVE_COLOR

      if (tabId == R.id.nav_myskin) {
          icon?.clearColorFilter()
          val ivMySkin = tab.findViewById<ImageView>(R.id.ivMySkin)
          val vMySkinShadow = tab.findViewById<View>(R.id.vMySkinShadow)
          
          // Cancel previous animators if any
          (ivMySkin?.getTag(R.id.nav_myskin) as? android.animation.Animator)?.cancel()
          (vMySkinShadow?.getTag(R.id.nav_myskin) as? android.animation.Animator)?.cancel()

          if (isActive) {
              ivMySkin?.setImageResource(R.drawable.ic_skin_mainbar)
              ivMySkin?.scaleX = 1f
              ivMySkin?.scaleY = 1f
              
              vMySkinShadow?.visibility = View.VISIBLE
              val shadowAnim = android.animation.ObjectAnimator.ofPropertyValuesHolder(
                  vMySkinShadow,
                  android.animation.PropertyValuesHolder.ofFloat(View.SCALE_X, 1.0f, 1.3f, 1.0f),
                  android.animation.PropertyValuesHolder.ofFloat(View.SCALE_Y, 1.0f, 1.3f, 1.0f),
                  android.animation.PropertyValuesHolder.ofFloat(View.ALPHA, 0.6f, 0.1f, 0.6f)
              ).apply {
                  duration = 2000
                  repeatCount = android.animation.ObjectAnimator.INFINITE
                  start()
              }
              vMySkinShadow?.setTag(R.id.nav_myskin, shadowAnim)
          } else {
              ivMySkin?.setImageResource(R.drawable.ic_skin_mainbar_nonactive)
              vMySkinShadow?.visibility = View.GONE
              vMySkinShadow?.scaleX = 1f
              vMySkinShadow?.scaleY = 1f
              
              val iconAnim = android.animation.ObjectAnimator.ofPropertyValuesHolder(
                  ivMySkin,
                  android.animation.PropertyValuesHolder.ofFloat(View.SCALE_X, 1.0f, 1.1f, 1.0f),
                  android.animation.PropertyValuesHolder.ofFloat(View.SCALE_Y, 1.0f, 1.1f, 1.0f)
              ).apply {
                  duration = 2000
                  repeatCount = android.animation.ObjectAnimator.INFINITE
                  start()
              }
              ivMySkin?.setTag(R.id.nav_myskin, iconAnim)
          }
          label?.setTextColor(Color.parseColor(color))
          label?.setTypeface(null, if (isActive) Typeface.BOLD else Typeface.NORMAL)
      } else {
          icon?.setColorFilter(Color.parseColor(color))
          if (isActive) {
              icon?.setBackgroundResource(R.drawable.shape_nav_active_bg)
          } else {
              icon?.background = null
          }
          label?.setTextColor(Color.parseColor(color))
          label?.setTypeface(null, if (isActive) Typeface.BOLD else Typeface.NORMAL)
      }
    }
  }

  fun showLoginRequiredDialog(context: android.content.Context) {
      val dialogView = android.view.LayoutInflater.from(context).inflate(R.layout.dialog_login_required, null)
      val dialog = androidx.appcompat.app.AlertDialog.Builder(context)
          .setView(dialogView)
          .create()
          
      val btnConfirm = dialogView.findViewById<android.view.View>(R.id.btnConfirmLogin)
      val btnCancel = dialogView.findViewById<android.view.View>(R.id.btnCancelLogin)
      
      btnConfirm.setOnClickListener {
          dialog.dismiss()
          val intent = android.content.Intent(context, com.veganbeauty.app.features.home.welcome.HomeWelcomeActivity::class.java)
          intent.putExtra("DIRECT_LOGIN", true)
          intent.flags = android.content.Intent.FLAG_ACTIVITY_NEW_TASK or android.content.Intent.FLAG_ACTIVITY_CLEAR_TASK
          context.startActivity(intent)
      }
      
      btnCancel.setOnClickListener {
          dialog.dismiss()
      }
      
      dialog.window?.setBackgroundDrawable(android.graphics.drawable.ColorDrawable(android.graphics.Color.TRANSPARENT))
      dialog.show()
  }

  fun navigate(fragment: Fragment, tabId: Int) {
    val isLoggedIn = com.veganbeauty.app.data.local.ProfileSession.isLoggedIn(fragment.requireContext())
    val target =
        when (tabId) {
          R.id.nav_home -> HomeFragment()
          R.id.nav_shop -> com.veganbeauty.app.features.shop.home.ShopHomeFragment()
          R.id.nav_myskin -> com.veganbeauty.app.features.myskin.MySkinFragment()
          R.id.nav_community -> if (isLoggedIn) com.veganbeauty.app.features.community.com_feed.ComLoadingFragment() else com.veganbeauty.app.features.community.beauty_hub.CommunityBeautyHubFragment()
          R.id.nav_account -> AccountProfileFragment()
          else -> null
        }

        target?.let {
          fragment.parentFragmentManager.beginTransaction()
              .replace(R.id.main_container, it)
              .commit()
        }
  }
}
